src/stk/common/stack_trace.h

```diff
@@ -2,32 +2,91 @@
 
 #include <sstream>
 
-#if defined(__GNUC__) && defined(__linux__)
+#if defined(__linux__) && defined(STK_USE_BACKWARD)
 
-// Relying on GNU extensions
+#include <backward.hpp>
+
+void get_stack_trace(std::ostringstream& s, const int skip = 0)
+{
+    s << std::endl;
+    backward::StackTrace st;
+    st.load_here(128);
+    st.skip_n_firsts(1 + skip); // skip current frame
+    backward::Printer p;
+    p.object = p.address = true;
+    p.color_mode = backward::ColorMode::always;
+    p.print(st, s);
+}
+
+#elif  defined(__GNUC__) && defined(__linux__)
+
+// Linux-only, relying on GNU extensions
 #include <execinfo.h>
 #include <unistd.h>
 #include <stdlib.h>
 
-void stack_trace(std::ostringstream& s)
+void get_stack_trace(std::ostringstream& s, const int skip = 0)
 {
     #define BT_BUF_SIZE 1024
     void *buffer[BT_BUF_SIZE];
     int bt_len = backtrace(buffer, BT_BUF_SIZE);
     char **bt = backtrace_symbols(buffer, bt_len);
     s << std::endl;
-    for (int i = 0; i < bt_len; ++i) {
+    for (int i = 1 + skip; i < bt_len; ++i) { // skip current frame
         s << std::string (bt[i]) << std::endl;
     }
     free(bt);
     #undef BT_BUF_SIZE
 }
 
-#else // defined(__GNUC__) && defined(__linux__)
+#elif defined(_WIN32)
+
+#include <process.h>
+#include <iostream>
+#include <sstream>
+#include <Windows.h>
+#include "dbghelp.h"
+
+#pragma comment ( lib, "dbghelp.lib" )
 
-// FIXME to implement something equivalent for MSVC
-void stack_trace(std::ostringsream& /* s */)
+int get_stack_trace(std::ostream& s, const int skip = 0)
```

Comment:
I tried to implement something that should work on Windows. Not a system in my programming comfort zone, though, so I hope this thing will actually work...

Reply:
Cool, seems to work without problems.